api/user/posts/posts.py

```diff
@@ -1,46 +1,220 @@
+    del_stmt = (
+        delete(Post)
+        .where(object_id=post_id, author_id=author.id, inbox=author_id)
+        .execution_options(synchronize_session="fetch")
+    )
+
+    db.session.execute(del_stmt)
+    db.session.commit()
 
 
 @posts_bp.route("/<string:author_id>/posts/<string:post_id>", methods=["PUT"])
 def create_post(author_id: str, post_id: str):
+    # TODO is this a new post or does it have existing comments???
     """create a post where its id is post_id"""
-    pass
+    response = {}
+    data = request.json
+
+    data = request.json
+    published = data.get("published")
+    title = data.get("title")
+    origin = data.get("origin")
+    source = data.get("source")
+    description = data.get("description", "")
+    contentType = data.get("contentType")
+    content = data.get("content")
+    categories = ",".join(data.get("categories"))
+    visibility = data.get("visibility")
+    unlisted = data.get("unlisted")
+    author_url = ""
+    try:
+        author = Author.query.filter_by(object_id=author_id).first()
+        post = Post(
+            id=post_id,
+            published=published,
+            title=title,
+            origin=origin,
+            source=source,
+            description=description,
+            content=content,
+            contentType=contentType,
+            categories=categories,
+            visibility=visibility,
+            unlisted=unlisted,
+            author_id=author_url,
+            inbox=author_id,
+        )
+
+        db.session.add(post)
+        db.session.commit()
+
+    except:
+        response["Success"] = 0
+        response["Message"] = "Author does not exist on this server."
+    else:
+        author_url = author.id
+        response["Success"] = 1
+        response["Message"] = "Post created succesfully."
+
+    return response
 
 
 @posts_bp.route("/<string:author_id>/posts", methods=["POST"])
 def create_post_auto_gen_id(author_id: str):
     """create a new post but generate a new id"""
-    pass
+    response = {}
+
+    data = request.json
+    published = data.get("published")
+    title = data.get("title")
+    origin = data.get("origin")
+    source = data.get("source")
+    description = data.get("description", "")
+    contentType = data.get("contentType")
+    content = data.get("content")
+    categories = ",".join(data.get("categories"))
+    visibility = data.get("visibility")
+    unlisted = data.get("unlisted")
+    author_url = ""
+    try:
+        author = Author.query.filter_by(object_id=author_id).first()
+        post = Post(
+            published=published,
+            title=title,
+            origin=origin,
+            source=source,
+            description=description,
+            content=content,
+            contentType=contentType,
+            categories=categories,
+            visibility=visibility,
+            unlisted=unlisted,
+            author_id=author_url,
+            inbox=author_id,
+        )
```

Comment:
As I see it, there's really no benefit to assinging to intermediate vars ehre, might as well make it direct:
```py
published=data.get("published")
```
Thoughts?